compass/landice/tests/ismip6_forcing/ocean_thermal/process_thermal_forcing.py

```diff
@@ -0,0 +1,319 @@
+import os
+import shutil
+import subprocess
+import xarray as xr
+from compass.landice.tests.ismip6_forcing.create_mapfile \
+    import build_mapping_file
+from mpas_tools.io import write_netcdf
+from compass.step import Step
+
+
+class ProcessThermalForcing(Step):
+    """
+    A step for creating a mesh and initial condition for dome test cases
+    """
+    def __init__(self, test_case):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        test_case : compass.landice.tests.ismip6_forcing.ocean_thermal.
+                    OceanThermal
+            The test case this step belongs to
+        """
+        super().__init__(test_case=test_case, name='process_thermal_forcing')
+
+    def setup(self):
+        """
+        Set up this step of the test case
+        """
+        config = self.config
+        section = config['ismip6_ais']
+        base_path_ismip6 = section.get('base_path_ismip6')
+        base_path_mali = section.get('base_path_mali')
+        mali_mesh_file = section.get('mali_mesh_file')
+        period_endyear = section.get("period_endyear")
+        model = section.get("model")
+        scenario = section.get("scenario")
+
+        section = config['ismip6_ais_ocean_thermal']
+        process_obs_data = section.getboolean("process_obs_data")
+
+        if period_endyear == "NotAvailable":
+            raise ValueError("You need to supply a user config file, which "
+                             "should contain the ismip6_ais "
+                             "section with the period_endyear option")
+        if model == "NotAvailable":
+            raise ValueError("You need to supply a user config file, which "
+                             "should contain the ismip6_ais "
+                             "section with the model option")
+        if scenario == "NotAvailable":
+            raise ValueError("You need to supply a user config file, which "
+                             "should contain the ismip6_ais "
+                             "section with the scenario option")
```

Comment:
This could be checked in the test case's `configure()` along with the other options (in a shared function).